File: task1/python/main.py
def min_operations(s):
    n = len(s)
    max_i = -1
    # Проверяем возможные позиции i, начиная с наибольшей
    for i in range((n // 2 - 1), -1, -1):
        substr_len = i + 1
        if 2 * substr_len > n:
            continue
        if s[:substr_len] == s[substr_len : 2 * substr_len]:
            max_i = i
            break  # Нашли максимальный подходящий i
    return n - max_i if max_i != -1 else n

#тут получаем строку от пользователя, обрабатываем, выводим
s = input("Введите строку: ")
print(min_operations(s))  

File: task1/kotlin/main.kt
fun minOperations(s: String): Int {
    val n = s.length
    var maxI = -1
    
    // Перебираем возможные длины подстроки, начиная с наибольшей
    for (i in (n / 2 - 1) downTo 0) {
        val substrLen = i + 1
        if (2 * substrLen > n) continue
        // Проверяем, совпадает ли первая подстрока с последующей
        if (s.substring(0, substrLen) == s.substring(substrLen, 2 * substrLen)) {
            maxI = i
            break
        }
    }
    // Возвращаем минимальное количество операций
    return if (maxI != -1) n - maxI else n
}

fun main() {
    //тут получаем строку от пользователя, обрабатываем, выводим
    print("Введите строку: ")
    val input = readLine()? ?: ""
    println("Результат: ${minOperations(input)}")
}
